app/assets/javascripts/template.questions.js

```diff
@@ -0,0 +1,126 @@
+var TemplateQuestions;
+var TopicQuestions;
+var TemplateID;
+var TemplateName;
+
+function add_tr(q){
+    var tr = document.getElementById("question_list").insertRow(-1);
+    tr.id = "question-row-" + q.id;
+    var tdc = tr.insertCell(0);
+    var tdd = tr.insertCell(1);
+    tdc.className = "name";
+    tdd.className = "del";
+    tdc.innerHTML = q.content;
+    var remove_button_id = "remove-question-button-" + q.id;
+    tdd.innerHTML = "<button id=\"" + remove_button_id + "\"class=\"btn btn-info remove-question-button\" type=\"button\" name=\"button\">Remove</button>";
+    $("#question_id option[value="+ q.id+"]").remove();
+    var found_question = $.grep(TemplateQuestions, function(e){ return e.id == q.id });
+    if (found_question.length == 0) {
+        TemplateQuestions.push(q);
+    }
+}
+
+function index_of_element(array, id){
```

Comment:
why not?
http://www.w3schools.com/jsref/jsref_indexof_array.asp
http://api.jquery.com/jQuery.grep/